emily/handler/tests/integration/deposit.rs

```diff
@@ -548,3 +548,114 @@ async fn update_deposits_updates_chainstate() {
+    let bitcoin_tx_output_index = 0;
+
+    // Setup test deposit transaction.
+    let DepositTxnData {
+        reclaim_script, deposit_script, ..
+    } = DepositTxnData::new(DEPOSIT_LOCK_TIME, DEPOSIT_MAX_FEE, DEPOSIT_AMOUNT_SATS);
+
+    let create_deposit_body = CreateDepositRequestBody {
+        bitcoin_tx_output_index,
+        bitcoin_txid: bitcoin_txid.into(),
+        deposit_script: deposit_script.clone(),
+        reclaim_script: reclaim_script.clone(),
+    };
+
+    apis::deposit_api::create_deposit(&configuration, create_deposit_body.clone())
+        .await
+        .expect("Received an error after making a valid create deposit request api call.");
+
+    let response = apis::deposit_api::get_deposit(
+        &configuration,
+        bitcoin_txid,
+        &bitcoin_tx_output_index.to_string(),
+    )
+    .await
+    .expect("Received an error after making a valid get deposit api call.");
+    assert_eq!(response.bitcoin_txid, bitcoin_txid);
+    assert_eq!(response.status, Status::Pending);
+
+    let mut fulfillment: Option<Option<Box<Fulfillment>>> = None;
+
+    if status == Status::Confirmed {
+        fulfillment = Some(Some(Box::new(Fulfillment {
+            bitcoin_block_hash: "bitcoin_block_hash".to_string(),
+            bitcoin_block_height: 23,
+            bitcoin_tx_index: 45,
+            bitcoin_txid: "test_fulfillment_bitcoin_txid".to_string(),
+            btc_fee: 2314,
+            stacks_txid: "test_fulfillment_stacks_txid".to_string(),
+        })));
+    }
+
+    apis::deposit_api::update_deposits(
+        &configuration,
+        UpdateDepositsRequestBody {
+            deposits: vec![DepositUpdate {
+                bitcoin_tx_output_index: bitcoin_tx_output_index,
+                bitcoin_txid: bitcoin_txid.into(),
+                fulfillment,
+                last_update_block_hash: "update_block_hash".into(),
+                last_update_height: 34,
+                status,
+                status_message: "foo".into(),
+            }],
+        },
+    )
+    .await
+    .expect("Received an error after making a valid update deposit request api call.");
+
+    let response = apis::deposit_api::get_deposit(
+        &configuration,
+        bitcoin_txid,
+        &bitcoin_tx_output_index.to_string(),
+    )
+    .await
+    .expect("Received an error after making a valid get deposit api call.");
+    assert_eq!(response.bitcoin_txid, bitcoin_txid);
+    assert_eq!(response.status, status);
+
+    if should_reject {
+        apis::deposit_api::create_deposit(&configuration, create_deposit_body)
+            .await
+            .expect_err(&format!(
+                "We should reject duplicate deposits, if old one is {:#?}",
+                status
+            ));
+    } else {
+        apis::deposit_api::create_deposit(&configuration, create_deposit_body)
+            .await
+            .expect("Received an error after making a valid get deposit api call.");
```

Comment:
nit: wrong err message, also we could simplify this block by `assert_eq` that `create_deposit(...).await.is_err()` matches `should_reject`.
Note that we already have the info related to which test case is running (eg, `emily-handler::integration deposit::overwrite_deposit::should_not_reject_pending_duplicate`), so we would still have everything we need to find the failing case.